waiter/src/waiter/process_request.clj

```diff
@@ -95,6 +95,10 @@
                              ;; cancel_stream_error is used to indicate that the stream is no longer needed
                              (and (instance? IOException error) (= "cancel_stream_error" error-message))
                              [:client-error "Client action means stream is no longer needed" http-400-bad-request error-class]
+                             ;; no_error is used to indicate that the stream is no longer needed
+                             ;; HTTP2 spec: The associated condition is not a result of an error...indicate graceful shutdown of a connection.
+                             (and (instance? IOException error) (= "no_error" error-message))
+                             [:generic-error "Bad request, server closed connection" http-400-bad-request error-class]
```

Comment:
If this is a graceful shutdown, why are we calling this a "Bad request"

Reply:
I classified this as a 400 generic error as we've observed this error so far for scenarios where some constraint has been violated, e.g. max size on grpc payload.

Reply:
I changed the error message to: `Server closed connection as stream no longer needed`